include/circt/Dialect/FIRRTL/Passes.td

```diff
@@ -23,4 +23,18 @@ def LowerFIRRTLTypes : Pass<"firrtl-lower-types", "firrtl::FModuleOp"> {
   let constructor = "circt::firrtl::createLowerFIRRTLTypesPass()";
 }
 
+def BlackboxMemory : Pass<"firrtl-blackbox-memory", "firrtl::CircuitOp"> {
+  let summary = "Replace all FIRRTL memories with an external module blackbox.";
+  let description = [{
+    This pass replaces all sequential memory operations with an external module
+    blackbox.  For each memory operation, it creates a FModuleOp which
+    replicates the return types of the memory operation, and replaces the
+    MemoryOp with an instance of the module.  Inside the new module, an
+    instance of an FExtModule blackbox is created.  The blackbox module must
+    use the same parameter naming conventions used by the ReplaceSeqMemories
+    pass in the Scala FIRRTL compiler.
+  }];
+  let constructor = "circt::firrtl::createBlacKBoxMemoryPass()";
+}
```

Comment:
There is some value doing this here, but we may lose the fact this was a memory when we hit rtl.  We don't have the generator nodes yet, but that's what we want the firrtlToRTL pass to lower to.  When we get there, we may need to not have done this.  I'm going to review this as is against the current tree with the understanding that we may need to move the functionality once the generator nodes exist.